Lib/fontTools/cffLib/specializer.py

```diff
@@ -304,7 +304,7 @@ def _convertBlendOpToArgs(blendList):
 	deltaArgs = args[numBlends:]
 	numDeltaValues = len(deltaArgs)
 	deltaList = [ deltaArgs[i:i + numRegions] for i in range(0, numDeltaValues, numRegions) ]
-	blend_args = [ a + b for a, b in zip(defaultArgs,deltaList)]
+	blend_args = [ a + b + [1] for a, b in zip(defaultArgs,deltaList)]
```

Comment:
why `[1]`, what's special about that magic numnber?

Reply:
In the generalizer, each blend operator has one blended number in it... Since this number is always fixed, Ariza didn't encode it. But that made the format of the arguments different between generalized commands and specialized commands.

Copying from your comment in the bug:

Original program:
```
  96 -12 -20 1 blend hmoveto 432 74 120 1 blend 660 -432 -74 -120 1 blend -660 hlineto 48 32 101 164 45 72 2 blend rmoveto 102 176 64 106 -47 -76 -75 -120 -17 -28 -6 -10 4 blend rlineto 4 hlineto 62 -106 100 -176 -16 -26 6 10 -46 -74 75 120 4 blend rlineto -332 126 204 1 blend hlineto 166 334 -64 -104 -9 -14 2 blend rmoveto -56 92 -94 168 10 16 3 4 47 76 -82 -132 4 blend rlineto 302 -113 -182 1 blend hlineto -94 -168 -54 -92 47 76 82 132 9 14 -3 -4 4 blend rlineto -4 hlineto -176 -292 9 14 25 40 2 blend rmoveto 536 -124 -200 1 blend vlineto 154 -270 -154 -266 -45 -72 63 102 45 72 61 98 4 blend rlineto 354 -17 -26 1 blend hmoveto -152 266 152 270 44 70 -61 -98 -44 -70 -63 -102 4 blend rlineto -536 124 200 1 blend vlineto
```
Generalized program:
```
  96 -12 -20 blend 0 rmoveto 432 74 120 blend 0 rlineto 0 660 rlineto -432 -74 -120 blend 0 rlineto 0 -660 rlineto 48 101 164 blend 32 45 72 blend rmoveto 102 -47 -76 blend 176 -75 -120 blend rlineto 64 -17 -28 blend 106 -6 -10 blend rlineto 4 0 rlineto 62 -16 -26 blend -106 6 10 blend rlineto 100 -46 -74 blend -176 75 120 blend rlineto -332 126 204 blend 0 rlineto 166 -64 -104 blend 334 -9 -14 blend rmoveto -56 10 16 blend 92 3 4 blend rlineto -94 47 76 blend 168 -82 -132 blend rlineto 302 -113 -182 blend 0 rlineto -94 47 76 blend -168 82 132 blend rlineto -54 9 14 blend -92 -3 -4 blend rlineto -4 0 rlineto -176 9 14 blend -292 25 40 blend rmoveto 0 536 -124 -200 blend rlineto 154 -45 -72 blend -270 63 102 blend rlineto -154 45 72 blend -266 61 98 blend rlineto 354 -17 -26 blend 0 rmoveto -152 44 70 blend 266 -61 -98 blend rlineto 152 -44 -70 blend 270 -63 -102 blend rlineto 0 -536 124 200 blend rlineto
```

You see the generalized program missing the `1` before the `blend`.

Reply:
In the generalizer, each blend operator has one blended number in it... Since this number is always fixed, Ariza didn't encode it. But that made the format of the arguments different between generalized commands and specialized commands.

Copying from your comment in the bug:

Original program:
```
  96 -12 -20 1 blend hmoveto 432 74 120 1 blend 660 -432 -74 -120 1 blend -660 hlineto 48 32 101 164 45 72 2 blend rmoveto 102 176 64 106 -47 -76 -75 -120 -17 -28 -6 -10 4 blend rlineto 4 hlineto 62 -106 100 -176 -16 -26 6 10 -46 -74 75 120 4 blend rlineto -332 126 204 1 blend hlineto 166 334 -64 -104 -9 -14 2 blend rmoveto -56 92 -94 168 10 16 3 4 47 76 -82 -132 4 blend rlineto 302 -113 -182 1 blend hlineto -94 -168 -54 -92 47 76 82 132 9 14 -3 -4 4 blend rlineto -4 hlineto -176 -292 9 14 25 40 2 blend rmoveto 536 -124 -200 1 blend vlineto 154 -270 -154 -266 -45 -72 63 102 45 72 61 98 4 blend rlineto 354 -17 -26 1 blend hmoveto -152 266 152 270 44 70 -61 -98 -44 -70 -63 -102 4 blend rlineto -536 124 200 1 blend vlineto
```
Generalized program:
```
  96 -12 -20 blend 0 rmoveto 432 74 120 blend 0 rlineto 0 660 rlineto -432 -74 -120 blend 0 rlineto 0 -660 rlineto 48 101 164 blend 32 45 72 blend rmoveto 102 -47 -76 blend 176 -75 -120 blend rlineto 64 -17 -28 blend 106 -6 -10 blend rlineto 4 0 rlineto 62 -16 -26 blend -106 6 10 blend rlineto 100 -46 -74 blend -176 75 120 blend rlineto -332 126 204 blend 0 rlineto 166 -64 -104 blend 334 -9 -14 blend rmoveto -56 10 16 blend 92 3 4 blend rlineto -94 47 76 blend 168 -82 -132 blend rlineto 302 -113 -182 blend 0 rlineto -94 47 76 blend -168 82 132 blend rlineto -54 9 14 blend -92 -3 -4 blend rlineto -4 0 rlineto -176 9 14 blend -292 25 40 blend rmoveto 0 536 -124 -200 blend rlineto 154 -45 -72 blend -270 63 102 blend rlineto -154 45 72 blend -266 61 98 blend rlineto 354 -17 -26 blend 0 rmoveto -152 44 70 blend 266 -61 -98 blend rlineto 152 -44 -70 blend 270 -63 -102 blend rlineto 0 -536 124 200 blend rlineto
```

You see the generalized program missing the `1` before the `blend`.